lib/plugin/merge-options.ts

```diff
@@ -3,11 +3,13 @@ import { isString } from '@nestjs/common/utils/shared.utils';
 export interface PluginOptions {
   typeFileNameSuffix?: string | string[];
   introspectComments?: boolean;
+  advancedTypeIntrospection?: boolean;
 }
 
 const defaultOptions: PluginOptions = {
   typeFileNameSuffix: ['.input.ts', '.args.ts', '.entity.ts', '.model.ts'],
   introspectComments: false,
+  advancedTypeIntrospection: true,
```

Comment:
this flag enables "old" behavior and aimed to eliminate most of breaking changes.